README.md

```diff
@@ -128,35 +61,182 @@ You are now ready to clone the repository and run `npm install`.
+```
+npm run dev
+```
 
-[Open Build Service]: https://build.opensuse.org/
 
-### How To Test
+## Tests
 
-Use the following commands to run unit tests and e2e tests.
+To run the unit tests:
 
 ```
 npm test
+```
+
+To run the integration tests:
+
+```
 npm run test:e2e
 ```
+
+
+## Building
+
+Rancher can be built from source on Windows, macOS or Linux.
+Cross-compilation is currently not supported.
+
+
+### Windows and macOS
+
+```
+npm run build
+```
+
+The build output goes to `dist/`.
+
+
+### Linux
+
+On Linux it is not possible to completely build Rancher Desktop from
+the development environment. This is because [Open Build Service][OBS]
+is used to build the application package into a variety of Linux
+package formats. However, you can access development builds for Linux.
+To learn how see below.
+
+[OBS]: https://build.opensuse.org/
+
+
+## Development Builds
+
+### Windows and macOS
+
+Each commit triggers a GitHub Actions run that results in application bundles
+(`.exe`s and `.dmg`s) being uploaded as artifacts. This can be useful if you
+want to test the latest build of Rancher Desktop as built by the build system.
+You can download these artifacts from the Summary page of completed `package`
+actions.
+
+
+### Linux
+
+Like with Windows and macOS, Linux builds of Rancher Desktop are made from each
+commit. However on Linux, only part of the process is done by GitHub Actions.
+The final part of it is done by [Open Build Service][OBS].
+
+There are two channels of the Rancher Desktop repositories: `dev` and `stable`.
+`stable` is the channel that most users use. It is the one that users are
+instructed to add in the official [documentation][docs], and the one that contains
+builds that are created from official releases. `dev` is the channel that we are
+interested in here: it contains builds created from the latest commit made on
+the `main` branch, and on any branches that match the format `release-*`. To
+learn how to install the development repositories, see below.
+
+When using the `dev` repositories, it is important to understand the format of
+the versions of Rancher Desktop available from the `dev` repositories.
+The versions are in the format:
+
+```
+<priority>.<branch>.<commit>
+```
+
+where:
+
+`priority` is a meaningless number that exists to give versions built from the `main`
+branch priority over versions built from the `release-*` branches when updating.
+
+`branch` is the branch name; dashes are removed due to constraints imposed by
+package formats.
+
+`commit` is the shortened hash of the commit used to make the build.
+
+[docs]: https://docs.rancherdesktop.io
+[OBS]: https://build.opensuse.org/
+
+
+#### `.deb` Development Repository
+
+You can add the repo with the following steps:
+
+```
+curl -s https://download.opensuse.org/repositories/isv:/Rancher:/stable/deb/Release.key | gpg --dearmor | sudo dd status=none of=/usr/share/keyrings/isv-rancher-stable-archive-keyring.gpg
```

Comment:
Interesting use of `dd`. I've mostly seen `sudo tee` in the same place.

Reply:
Yeah I had never seen it used in that way either, until I came across a SO comment that did this. But with `dd status=none` doesn't print out the path, whereas `tee` does - I figured that less output would be better.